crates/grafbase-sdk/src/types/token.rs

```diff
@@ -0,0 +1,41 @@
+use serde::Serialize;
+
+use crate::{cbor, wit, SdkError};
+
+/// A structure representing an authentication token
+pub struct RawToken(wit::RawToken);
+
+impl From<RawToken> for wit::RawToken {
+    fn from(token: RawToken) -> wit::RawToken {
+        token.0
+    }
+}
+
+impl RawToken {
+    /// Anonymous token, without any data associated with it.
+    pub fn anonymous() -> Self {
+        Self(wit::RawToken::Anonymous)
+    }
+
+    /// Token with arbitrary data associated with it. If the token you generate is meant to be
+    /// usable by other extensions, it's recommended to use either `json_claims` or `cbor_claims`.
+    pub fn bytes(bytes: Vec<u8>) -> Self {
+        Self(wit::RawToken::Bytes(bytes))
+    }
+
+    /// Serialize a set of claims as the token.
+    pub fn claims<T: Serialize>(claims: T) -> Result<Self, SdkError> {
+        let bytes = cbor::to_vec(&claims)?;
+        Ok(Self(wit::RawToken::Claims(wit::Data::Cbor(bytes))))
+    }
+
+    /// Token with claims serialized in JSON.
+    pub fn json_claims(claims: Vec<u8>) -> Self {
+        Self(wit::RawToken::Claims(wit::Data::Json(claims)))
+    }
+
+    /// Token with claims serialized in CBOR.
+    pub fn cbor_claims(claims: Vec<u8>) -> Self {
+        Self(wit::RawToken::Claims(wit::Data::Cbor(claims)))
+    }
```

Comment:
Should these be public?

Reply:
removed those, went back to a simpler variant with only bytes & anonymous case